ape_safe/accounts.py

```diff
@@ -36,32 +47,94 @@ def aliases(self) -> Iterator[str]:
         for p in self._account_files:
             yield p.stem
 
-    def __len__(self) -> int:
-        return len([*self._account_files])
-
     @property
     def accounts(self) -> Iterator[AccountAPI]:
         for account_file in self._account_files:
-            yield SafeAccount(account_file_path=account_file)  # type: ignore
+            yield SafeAccount(account_file_path=account_file)
+
+    def __len__(self) -> int:
+        return len([*self._account_files])
+
+    def __setitem__(self, alias: str, address: str):
+        self.save_account(alias, address)
+
+    def __delitem__(self, alias: str):
+        self.delete_account(alias)
 
     def save_account(self, alias: str, address: str):
         """
         Save a new Safe to your ape configuration.
+
+        Args:
+            alias (str): The alias to save the Safe under.
+            address (str): The address of the Safe account.
         """
         chain_id = self.provider.chain_id
         account_data = {"address": address, "deployed_chain_ids": [chain_id]}
-        path = self.data_folder.joinpath(f"{alias}.json")
+        path = self._get_path(alias)
+        if path.is_file():
+            raise ApeSafeError(f"Safe with alias '{alias}' already exists.")
+
         path.write_text(json.dumps(account_data))
 
     def load_account(self, alias: str) -> "SafeAccount":
-        account_path = self.data_folder.joinpath(f"{alias}.json")
+        """
+        Load the Safe account.
+
+        Args:
+            alias (str): The alias the Safe account is saved under.
+
+        Returns:
+            ``:class:~ape_safe.accounts.SafeAccount``: The Safe account loaded.
+        """
+        account_path = self._get_path(alias)
+        if not account_path.is_file():
+            raise ApeSafeError(f"Safe with '{alias}' does not exist")
+
         return SafeAccount(account_file_path=account_path)
 
     def delete_account(self, alias: str):
-        path = self.data_folder.joinpath(f"{alias}.json")
+        """
+        Delete the local Safe account.
+        **NOTE**: If the account does not exist, nothing happens.
+
+        Args:
+            alias (str): The alias the Safe account is saved under.
+        """
+        self._get_path(alias).unlink(missing_ok=True)
+
+    def _get_client(self, key: str) -> BaseSafeClient:
+        if key in self.aliases:
+            safe = self.load_account(key)
+            return safe.client
+
+        elif key in self:
+            account = self[key]  # type: ignore[index]
+            if not isinstance(account, SafeAccount):
+                raise TypeError("Safe container returned non-safe account.")
+
+            return account.client
+
+        else:
+            # Is not locally managed.
+            address = self.conversion_manager.convert(key, AddressType)
```

Comment:
**issue**: What if `self.conversion_manager.convert(key, AddressType)` is in `self`? Meaning if you give it an ENS? I am not sure it would hit the `elif` case above, or at least it seems like it wouldn't. Maybe that it doesn't matter currently because it will still *work*. However, I can foresee issues with assuming this control path is *not locally managed* as the comment suggest.

**suggestion**:

One possible solution is to change this else statement like this:

(first make helper method to share with the `elif` logic above:

```python
def _get_safe_by_address(self, address: AddressType) -> SafeAccount:
    safe = self[account]
    if not isinstance(account, SafeAccount):
        raise TypeError("Safe container returned non-safe account.")

    return account.client
```

And then the `elif` and `else` here kind of like:

```python
elif key in self:
    return self._get_safe_by_address(key)

else:
    address = self.conversion_manager.convert(key, AddressType)
    if address in self:
        return self._get_safe_by_address(address)
    
    # Is not locally managed.
    return SafeClient(address=address, chain_id=self.chain_manager.provider.chain_id
```

Notice now it will check if locally managed again of the conversion, and share the logic for doing so with the `elif`. And only after that part does the code assume the safe is not managed locally.

Reply:
fixed